integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/cache/CarbonShowCacheCommand.scala

```diff
@@ -41,25 +42,42 @@ import org.apache.carbondata.spark.util.CarbonScalaUtil
 import org.apache.carbondata.spark.util.CommonUtil.bytesToDisplaySize
 
 
-case class CarbonShowCacheCommand(tableIdentifier: Option[TableIdentifier],
+case class CarbonShowCacheCommand(showExecutorCache: Boolean,
+    tableIdentifier: Option[TableIdentifier],
     internalCall: Boolean = false)
   extends MetadataCommand {
 
   private lazy val cacheResult: Seq[(String, Int, Long, String)] = {
-    executeJobToGetCache(List())
+    executeJobToGetCache(showExecutorCache, List())
   }
 
   private val LOGGER = LogServiceFactory.getLogService(classOf[CarbonShowCacheCommand].getName)
 
   override def output: Seq[AttributeReference] = {
     if (tableIdentifier.isEmpty) {
-      Seq(
-        AttributeReference("Database", StringType, nullable = false)(),
-        AttributeReference("Table", StringType, nullable = false)(),
-        AttributeReference("Index size", StringType, nullable = false)(),
-        AttributeReference("Datamap size", StringType, nullable = false)(),
-        AttributeReference("Dictionary size", StringType, nullable = false)(),
-        AttributeReference("Cache Location", StringType, nullable = false)())
+      val isDistributedPruningEnabled = CarbonProperties.getInstance()
+        .isDistributedPruningEnabled("", "")
+      if (showExecutorCache == false) {
+        Seq(
+          AttributeReference("Database and Table", StringType, nullable = false)(),
+          AttributeReference("Index size", StringType, nullable = false)(),
+          AttributeReference("Datamap size", StringType, nullable = false)(),
+          AttributeReference("Dictionary size", StringType, nullable = false)(),
+          AttributeReference("Cache Location", StringType, nullable = false)())
+      } else {
+        if (!isDistributedPruningEnabled) {
+          Seq(
+            AttributeReference("Database and Table", StringType, nullable = false)(),
```

Comment:
Change "Database and Table" to "Identifier"

Reply:
done